web/src/views/User/UserPage.vue

```diff
@@ -29,8 +30,13 @@ export default defineComponent({
       { text: 'Admin', value: 'is_admin', sortable: false },
     ];
 
-    const users = usePaginatedResults(ref([]), api.getAllUsers, ref([]), itemsPerPage);
+    async function getUsers(params: SearchParams): Promise<SearchResponse<User>> {
+      return api.getAllUsers(params, search_filter.value);
+    }
+
+    const users = usePaginatedResults(ref([]), getUsers, ref([]), itemsPerPage);
     watch(options, () => users.setPage(options.value.page), { deep: true });
+    watch(search_filter, () => users.setPage(options.value.page), { deep: true });
```

Comment:
I'm a little bit confused about what this watcher is doing. Please let me know your thoughts on this.

As an aside, in Vue 3 we'd be able to use [one watcher for both of these refs](https://vuejs.org/guide/essentials/watchers.html#watch-source-types), but for now we're still on Vue 2 so this is the cleanest way to do it.

Reply:
I didn't have that in at first, but without it the `users` var wouldn't see changes made to the textbox and update itself.

Reply:
Should we boot the user back to page 1 when the filter changes? I'm wondering what the expecting behavior would be if someone was like on page 5 and then started typing in some filter.
```suggestion
    watch(search_filter, () => users.setPage(1), { deep: true });
```
(I actually don't know if the code indexes pages from 0 or 1, so this suggestion might be wrong)

Reply:
Good catch, I was curious what would happen if you were on page 5. It displays all the correct results but keeps you on that page and still lets you change pages even if there's only 1 page worth of results which is kind of funky. 

I added the change (it does index from 0)